base_classes/nyaml/NXsample_component.yaml

```diff
@@ -0,0 +1,148 @@
+category: base
+doc: |
+  One group like this per component can be recorded For a sample consisting of multiple components.
+symbols:
+  doc: |
+    symbolic array lengths to be coordinated between various fields
+  n_Temp: |
+    number of temperatures
+  n_eField: |
+    number of values in applied electric field
+  n_mField: |
+    number of values in applied magnetic field
+  n_pField: |
+    number of values in applied pressure field
+  n_sField: |
```

Comment:
flesh out the individual external simuli into own base classes:
NXexternal_stimulus

then:
NXelectric_field
NXmagnetic_field
NXthermal_field
NXmechanical_field
are instances of NXexternal_stimulus
then move the respective descriptors and characteristics into these instances

Reply:
Agreed. Add NXpressure, NXgas_phase_environment.

One tricky thing is temperature: it is actually a property of the sample, but not necessarily inside the sample itself. For example, in photoemission, you often measure the temperature of the sample holder (with a thermocouple) which is directly in contact with the sample.